app/permissions/repository.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+Canaid::Permissions.register_for(Repository) do
+  # repository: update, delete
+  can :manage_repository do |user, repository|
+    user.is_admin_of_team?(repository.team)
+  end
+
+  # repository: create/import record
+  can :create_repository_rows do |user, repository|
+    if user.teams.include?(repository.team)
+      user.is_normal_user_or_admin_of_team?(repository.team)
+    elsif (write_team_repos = repository
+                                .team_repositories
+                                .where(team_id: user.teams.pluck(:id))
```

Comment:
Better use '.where(team: user.teams)' - it will use subquery instead 2 separate ones like here